packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/notification/NotificationController.java

```diff
@@ -0,0 +1,65 @@
+package software.uncharted.terarium.hmiserver.controller.notification;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.media.ArraySchema;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.annotation.Secured;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import software.uncharted.terarium.hmiserver.models.notification.NotificationGroup;
+import software.uncharted.terarium.hmiserver.security.Roles;
+import software.uncharted.terarium.hmiserver.service.CurrentUserService;
+import software.uncharted.terarium.hmiserver.service.notification.NotificationService;
+
+@RequestMapping("/notification")
+@RestController
+@Slf4j
+@RequiredArgsConstructor
+public class NotificationController {
+
+    private final CurrentUserService currentUserService;
+    private final NotificationService notificationService;
+
+    @GetMapping("/user")
+    @Secured(Roles.USER)
+    @Operation(summary = "Return all recent notification groups for a user")
+    @ApiResponses(
+            value = {
+                @ApiResponse(
+                        responseCode = "200",
+                        description = "Returned recent notifications successfully",
+                        content =
+                                @Content(
+                                        array =
+                                                @ArraySchema(
+                                                        schema =
+                                                                @io.swagger.v3.oas.annotations.media.Schema(
+                                                                        implementation = NotificationGroup.class)))),
+                @ApiResponse(
+                        responseCode = "500",
+                        description = "There was an issue fetching the notifications",
+                        content = @Content)
+            })
+    public ResponseEntity<List<NotificationGroup>> getNotificationGroups(
+            @RequestParam(value = "since", required = false, defaultValue = "2") final long sinceInHours) {
```

Comment:
Why are we limiting ourselves to notification group less than 2 hours?

Reply:
No reason, just an arbitrary filtering value. I can bump it to 48 hours or something more reasonable.

Reply:
Thanks, just pure curiosity on my part